pipeline/scripts/15-dIEM_violin.R

```diff
@@ -0,0 +1,470 @@
+# For untargeted metabolomics, this tool calculates probability scores for 
+# metabolic disorders. In addition, it provides visual support with violin plots 
+# of the DIMS measurements for the lab specialists.
+# Input needed: 
+# 1. Excel file in which metabolites are listed with their intensities for
+#    controls (with C in samplename) and patients (with P in samplename) and their
+#    corresponding Z-scores. 
+# 2. All files from github: https://github.com/UMCUGenetics/dIEM
+
+library(dplyr) # tidytable is for other_isobaric.R (left_join)
+library(reshape2) # used in prepare_data.R
+library(openxlsx) # for opening Excel file
+library(ggplot2) # for plotting
+library(gridExtra) # for table top highest/lowest
+
+# load functions
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/same_samplename.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_data.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_data_perpage.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_toplist.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/violin_plots.R")
+#source("/hpc/dbg_mz/production/DIMS/pipeline/scripts/AddOnFunctions/prepare_alarmvalues.R")
+#temporary:
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/same_samplename.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_data.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_data_perpage.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_toplist.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/violin_plots.R")
+source("/hpc/dbg_mz/development/DIMS_Violinplots/pipeline/scripts/AddOnFunctions/prepare_alarmvalues.R")
+
+# define parameters - check after addition to run.sh
+cmd_args <- commandArgs(trailingOnly = TRUE)
+for (arg in cmd_args) {
+  cat("  ", arg, "\n", sep = "")
+}
+
+outdir   <- cmd_args[1] 
+run_name <- cmd_args[2] 
+z_score  <- as.numeric(cmd_args[3]) # use Z-scores (1) or not (0)? 
+
+# The list of parameters can be shortened for HPC. Leave for now.
+top_nr_IEM       <- 5 # number of diseases that score highest in algorithm to plot
+threshold_IEM    <- 5 # probability score cut-off for plotting the top diseases
+ratios_cutoff    <- -5 # z-score cutoff of axis on the left for top diseases
+nr_plots_perpage <- 20 # number of violin plots per page in PDF
+
+# Settings from config.R
+# binary variable: run function, yes(1) or no(0). Can be removed at later stage
+if (z_score == 1) { algorithm <- ratios <- violin <- 1 } #
```

Comment:
For readability, add newlines between if/else blocks.

Reply:
Agreed, newlines added.
This if statement needs to be removed at a later stage.